### PR TITLE
fix: unable to edit proxy if ratelimit is not set or <= 0

### PR DESCRIPTION
**Pull request by Kirari04**
this fix checks the ratelimit value only if the
requireRateLimit is set to true else it will use
the provided ratelimit value unless it is less or equal to 0 then it will default to 1000 (the same value as set inside the ui)

Previous Issue:
Screenshot from 2024-06-24 22-53-48

### CODE REVIEW

**Comment by Kirari04** [comment]
The second commit was by accident.
I wanted to open a new PR for that one :cry:

**Comment by tobychui** [comment]
Well no worry, it is just a few lines fix I can modify them directly in my v3.0.8 branch. However for the docker part, I guess you will need to open another PR for that and notify the docker maintainer for the update.

**Comment by tobychui** [comment]
@PassiveLemon it seems there are changes to the docker file in this PR. You mind take a look if I can merge this?

File: docker/Dockerfile [hunk 1]
```diff
@@ -8,16 +8,28 @@ RUN mkdir -p /opt/zoraxy/source/ &&\
...
-
 # If you build it yourself, you will need to add the src directory into the docker directory.
 COPY ./src/ /opt/zoraxy/source/
 
 WORKDIR /opt/zoraxy/source/
 
 RUN go mod tidy &&\
-    go build -o /usr/local/bin/zoraxy &&\
-    rm -r /opt/zoraxy/source/
+go build -o /usr/local/bin/zoraxy &&\
```
Comment (PassiveLemon) [inline docker/Dockerfile]: Slight nit:
Would you mind tabbing these? Just for continuity. Same further down.

Comment (Kirari04) [reply docker/Dockerfile]: Sure, i'll do that 👍

Comment (Kirari04) [reply docker/Dockerfile]: Sorry for the long wait.
I now formated the Dockerfile @PassiveLemon

**Comment by PassiveLemon** [comment]
LGTM.

**Review by tobychui — APPROVED** [review]
I have manually moved the merge for reverse proxy to the new branch. And the docker part should be ok to merge.

**Pull request merged**